sls-versions/src/main/java/com/palantir/sls/versions/VersionComparator.java

```diff
@@ -16,95 +16,53 @@
 
 package com.palantir.sls.versions;
 
-import static com.palantir.logsafe.Preconditions.checkArgument;
-
-import com.palantir.logsafe.SafeArg;
 import java.util.Comparator;
-import java.util.OptionalInt;
 
 /** Compares {@link OrderableSlsVersion}s by "newness", i.e., "1.4.0" is greater/newer/later than "1.2.1", etc.. */
 public enum VersionComparator implements Comparator<OrderableSlsVersion> {
```

Comment:
Fwiw, at this point, this entire comparator could simply be:
```
Comparator.comparingInt(OrderableSlsVersion::getMajorVersionNumber)
                .thenComparingInt(OrderableSlsVersion::getMinorVersionNumber)
                .thenComparingInt(OrderableSlsVersion::getPatchVersionNumber)
                .thenComparingInt(version -> version.rcVersionNumber().orElse(Integer.MAX_VALUE))
                .thenComparingInt(version -> version.snapshotVersionNumber().orElse(Integer.MIN_VALUE));
```
but I'm not sure if this would have a performance cost, and I can't make the `VersionComparator.INSTANCE` _be_ this either